tchannel/tornado/peer.py

```diff
@@ -430,7 +430,7 @@ def send(
                 )
         except Exception as e:
             # event: on_exception
-            self.tchannel.event_emitter.fire(
+            yield self.tchannel.event_emitter.fire(
                 EventType.on_exception, request, e,
             )
             raise
```

Comment:
this will not work - the exception will be lost after the yield.
you can save the `sys.exc_info()` before yielding and raise it after, but there are some subtleties with old vs new style exception-raising (i.e. `raise Exception, 'message'` vs `raise Exception('message')`, which means you can't just `raise exc[0], exc[1], exc[2]` in all cases.

Reply:
Good catch. Will fix.